superset/assets/src/SqlLab/components/QueryAutoRefresh.jsx

```diff
@@ -41,10 +41,18 @@ class QueryAutoRefresh extends React.PureComponent {
     const { queries, queriesLastUpdate } = this.props;
     const now = new Date().getTime();
 
+    // due to a race condition, queries can be marked as successful before the
+    // results key is set; this is a workaround until we fix the underlying
+    // problem
+    const isQueryRunning = q => (
+      ['running', 'started', 'pending', 'fetching'].indexOf(q.state) >= 0 ||
+      (q.state === 'success' && q.resultsKey === null)
```

Comment:
@DiggidyDave this logic is problematic. Sql Lab can run async and synchronous queries. For synchronous queries, it will not never have resultsKey stored in query table.

the new logic will make extra polling happen when user has at least 1 tab has synchronous query (and < 6 hours old).

Reply:
Ah, my bad. I'll fix it.